src/adr/0055-rspamd-spam-filtering.md

```diff
@@ -0,0 +1,66 @@
+# 55. RSPAMD
+
+Date: 2022-08-10
+
+## Status
+
+Accepted (lazy consensus).
+
+Implemented. 
+
+## Context
+
+In order to anti-spam, James provide 3 mechanism: 
+
+- SMTP Hook: decide to reject or not before spooling on the SMTP level
+- Mailet: custom the mailet pipeline for changing the mail processing when detect spam mail.
+- Mailbox listener:  based on user action, eg: move in/out the message to Inbox/Spam mailbox, then decide report spam or not to separate Spam filtering system.
+
+For basic, we can base on username, DSN filter, domains, IP... for configuring black or grey list.
+If we want to analysis deeper in each message content more, we need a more complex system.
+The current, James supports the [SpamAssassin](https://spamassassin.apache.org/) for this problem.
+
+For more selection, James's repository provides the RSpamD extension, which provides the same way as SpamAssassin but another system - [RSpamD](https://github.com/rspamd/rspamd)
```

Comment:
We should explain that we found hard to operate SpamAssassin, which we had performance issue with.